OptimizelySDK/Optimizely.cs

```diff
@@ -308,5 +296,226 @@ public Variation GetForcedVariation(string experimentKey, string userId)
+                variableValue = featureVariableUsageInstance.Value;
+                Logger.Log(LogLevel.INFO,
+                    $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureFlag.Key}"".");
+            }
+            else
+            {
+                Logger.Log(LogLevel.INFO,
+                    $@"User ""{userId}"" is not used in any variation for feature flag ""{featureFlag.Key}"", returning default value ""{variableValue}"".");
+            }
+
+            return variableValue;
+        }
+
+        /// <summary>
+        /// Gets boolean feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>bool | Feature variable value or null</returns>
+        public bool? GetFeatureVariableBoolean(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
+        {
+            var variableValue = GetFeatureVariableValueForType(featureKey, variableKey, userId,
+                userAttributes, FeatureVariable.VariableType.BOOLEAN);
+            
+            if (variableValue != null && Boolean.TryParse(variableValue, out bool booleanValue))
+                return booleanValue;
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets double feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>double | Feature variable value or null</returns>
+        public double? GetFeatureVariableDouble(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
+        {
+            var variableValue = GetFeatureVariableValueForType(featureKey, variableKey, userId,
+                userAttributes, FeatureVariable.VariableType.DOUBLE);
+            
+            if (variableValue != null && Double.TryParse(variableValue, out double doubleVaue))
+                return doubleVaue;
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets integer feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>int | Feature variable value or null</returns>
+        public int? GetFeatureVariableInteger(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
```

Comment:
Also log warning when unable to parse. See doc/Java/PHP

Reply:
Done